### PR TITLE
test deprecated - model

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Testing deprecated model hooks

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
it seems that it is failing on different unrelated test:
```
metrics = {'val_acc': 0.2395833283662796, 'val_loss': 3.2839901447296143}
step = 0

    def decorated(metrics, step):
        if "val_acc" in metrics:
>           assert step == logged_step
E           assert 0 == 1
```

**Comment by Borda** [comment]
@PyTorchLightning/core-contributors any suggestion why the logger does not increase steps?

**Comment by Borda** [comment]
ok, fixed the one above, but another occurred:
```
>           type(self).__name__, name))
--
1922 | E       AttributeError: 'LightningDataParallel' object has no attribute 'validation_end'
```

File: pytorch_lightning/trainer/evaluation_loop.py [hunk 1]
```diff
@@ -291,24 +291,22 @@ def evaluate(self, model, dataloaders, max_batches, test_mode: bool = False):
             outputs = outputs[0]
 
         # give model a chance to do something with the outputs (and method defined)
-        model = self.get_model()
-
-        if test_mode and self.is_overriden('test_epoch_end'):
-            eval_results = model.test_epoch_end(outputs)
-        elif self.is_overriden('validation_epoch_end'):
-            eval_results = model.validation_epoch_end(outputs)
+        model = self.get_model()  # if you call evaluate on given model, ths getter gives None
```
Comment (neggert) [inline pytorch_lightning/trainer/evaluation_loop.py]: This gets all the tests passing for me:

```suggestion
        if isinstance(model, (LightningDistributedDataParallel, LightningDataParallel)):
            model = model.module
```

(you also need to add the appropriate imports)

**Pull request merged**